### DISCUSSION TITLE
Handbell support

### DISCUSSION THREAD

**Discussion by lotaheart** [Audiveris/audiveris#730] [General]
I was pleased to see the development of drum sets and thought this might allow the import of handbell scores. Typically they have two noteheads, normal circles noteheads and diamond noteheads. I tried to import the attached file and Audiveris was able to find the quarter note diamonds but not the half note diamonds. However, the data file did not keep theses as diamonds but translated the quarter not diamonds it found to circle noteheads. Am I using the right version and should this capability be working?
![Bell Score Test](https://github.com/Audiveris/audiveris/assets/167720508/cea015b9-782e-4c6c-8e8e-280d26a45b71)

**Comment by hbitteur** [comment]
@lotaheart
Please forgive me for the late answer.

[I suppose you have selected the item "5-line unpitched percussion staves" in the book parameters]
I tried to process your example.

- The void diamonds (half notes) are not recognized.
- Most of the filled diamonds (quarter notes) are correctly recognized. I noticed only a couple of them mistaken for half-notes diamonds or circled-cross (in measure 25), all the other ones seem OK.

The main problem is with the void diamonds. 
I suspect the reason is their shape is much too faint:
![image](https://github.com/Audiveris/audiveris/assets/22053634/fc16bb79-df5b-409c-89a6-a1cadd1fed3f)
Too far from the expected shape (scaling put aside):
![image](https://github.com/Audiveris/audiveris/assets/22053634/414f993e-e71f-45e3-94a8-52fb43841f01)

For better results, we would need a musical font closer to your case. The font used by default is the `Bravura` font, you can select another one in the book parameters. I tried some but the results were not better, perhaps even worse.

If you know of some more suitable musical font, please let us know.

**Comment by lotaheart** [comment]
Thanks for the information. I am certainly a novice at using the Audiveris software.

I think the Leland font is much closer to the handbell music I normally see. Is that font available to load? When I processed the scanned image using the Musical Symbols font it seemed to give me the best results. Also, instead of the diamond note head, the mi note head is a closer notehead to the handbell music. Can the notehead processor be switched up to look for mi note heads instead of diamond noteheads?

**Comment by hbitteur** [comment]
I made a mistake.

The drum notation is available only on **percussion** staves (the ones that start with a **percussion** clef). On these unpitched staves, an intrument is defined by a specific shape (diamond, triangle, ellipsis, ...) located on a specific line/space position.

In your case, you have "**standard**" staves starting with **treble** or **bass** clefs, where the pitch is indicated by the line position. On these staves, Audiveris supports standard elliptic note heads. In addition, it can also support cross heads, provided you have selected this feature in the book parameters.
On these staves, Audiveris does not expect such shapes as the void/filled diamonds.
I never saw these shapes on "standard" staves, could you tell me what they mean?

**Comment by lotaheart** [comment]
Thanks again for your continued help. This is an example of music for handbell choirs. Handbell music will often have music for traditional bells (playing the standard elliptical head notes) and hand chimes (playing the diamond head notes) mixed in. Players will switch instruments often based on the note head type. Reading in these diamond shaped notes as diamonds is thus important to playing the music correctly.
